test/gurka/test_truck_restrictions.cc

```diff
@@ -0,0 +1,66 @@
+#include "gurka.h"
+#include <boost/format.hpp>
+#include <gtest/gtest.h>
+
+using namespace valhalla;
+
+class TruckRestrictionTest : public ::testing::TestWithParam<std::pair<std::string, std::string>> {
+protected:
+  static gurka::map map;
+
+  static void SetUpTestSuite() {
+    constexpr double gridsize = 10;
+
+    const std::string ascii_map = R"(
+      A---B---C---D
+    )";
+
+    const gurka::ways ways = {
+        {"AB", {{"highway", "residential"}}},
+        // all restrictions should be higher than our defaults, so we can actually see the impact of
+        // any single one
+        {"BC",
+         {{"highway", "residential"},
+          {"maxheight", "5"},
+          {"maxlength", "25"},
+          {"maxwidth", "3"},
+          {"hazmat", "no"},
+          {"maxaxles", "8"},
+          {"maxaxleload", "10"}}},
```

Comment:
I think truck restrictions weren't tested before, so I added a test for all of them. The map allows for Valhalla's truck defaults (4.11 m height etc) so we can verify that it's actually fine if no `costing_options` are set.